src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,501 @@
+SOLUTION_TEMPLATE_START =\
+    """schema: \"apigear.solution/1.0\"
+    name: {}
+    version: \"1.0\"
+
+    layers:
+    """
+SOLUTION_TEMPLATE_LAYER_UNREAL =\
+    """  - name: unreal
+        inputs:
+        - {module_name}
+        output: {output_path}
+        template: apigear-io/template-unreal
+    """
+SOLUTION_TEMPLATE_LAYER_CPP =\
+    """  - name: cpp
+        inputs:
+        - {module_name}
+        output: {output_path}
+        template: apigear-io/template-cpp14
+        features:
+        - examples_olink
+    """
+SOLUTION_TEMPLATE_LAYER_QT5 =\
+    """  - name: qt5
+        inputs:
+        - {module_name}
+        output: {output_path}
+        template: apigear-io/template-qt5
+    """
+SOLUTION_TEMPLATE_LAYER_QT6 =\
+    """  - name: qt6
+        inputs:
+        - {module_name}
+        output: {output_path}
+        template: apigear-io/template-qtcpp
+    """
+MODULE_FILENAME = "{}.module.yaml"
+MODULE_TEMPLATE_START =\
+    """schema: apigear.module/1.0
+    name: {}
+    version: \"1.0\"
+
+    """
+
+
+class SolutionLayers(Flag):
+    UNREAL = auto()
+    CPP = auto()
+    QT5 = auto()
+    QT6 = auto()
+
+
+class ApiGearBasicType(Enum):
+    BOOL = "bool"
+    INT32 = "int32"
+    INT64 = "int64"
+    FLOAT32 = "float32"
+    FLOAT64 = "float64"
+    STRING = "string"
+
+
+class ApiGearComplexType(abc.ABC):
+    def __init__(self, name):
+        super().__init__()
+
+        self.name = name
+
+
+class ApiGearStructure(ApiGearComplexType):
+    def __init__(self, name: str):
+        super().__init__(name)
+
+        self.properties: typing.Dict[str, ApiGearProperty] = {}
+
+
+class ApiGearEnumeration(ApiGearComplexType):
+    def __init__(self, name: str):
+        super().__init__(name)
+
+        # TODO: allow using custom values for variants?
+        self.variants: typing.List[str] = []
+
+
+class ApiGearType():
+    def __init__(self, type: str):
+        super().__init__()
+
+        self.is_array = False
+        self.type = type
+
+    @classmethod
+    def array(cls, type: str):
+        t = cls(type)
+        t.is_array = True
+        return t
+
+
+class ApiGearProperty():
+    def __init__(self, type: ApiGearType):
+        super().__init__()
+
+        self.type = type
+        self.description: typing.Optional[str] = None
+
+
+class ApiGearInterface:
+    def __init__(self, name: str):
+        super().__init__()
+
+        self.name: str = name
+        self.properties: typing.Dict[str, ApiGearProperty] = {}
+
+        # TODO: implement handling for operations
+        # self.operations
+
+        # TODO: implement handling for signals
+        # self.signals
+
+
+class ApiGearModule:
+    def __init__(self):
+        super().__init__()
+
+        self.interfaces: typing.List[ApiGearInterface] = []
+        self.enumerations: typing.List[ApiGearEnumeration] = []
+        self.structures: typing.List[ApiGearStructure] = []
+
+
+def get_allowed_enum_literal(name: str):
```

Comment:
please add typing hints for return values aswell

Reply:
Added typing hint sfor return value.